src/DropdownToggle.js

```diff
@@ -67,7 +67,7 @@ class DropdownToggle extends React.Component {
     ), cssModule);
     const children = props.children || <span className="sr-only">{ariaLabel}</span>;
 
-    if (nav) {
+    if (nav && Tag === Button) {
```

Comment:
I would eventually like to use an env var similar to react to remove proptypes in production code since they aren't used. 

> I was thinking about checking if Tag was a function and whether or not Tag.propTypes.tag was defined or something to see if it could accept this prop.

To avoid the need for that check I think we should switch the `<Tag ... />` to `<Button ... tag={UserTag} />` 

For components that kind of extend an existing component, like this one does with `Button`, I think the render method should always use the extended component and pass the tag prop along to that component. This way we never pass a reactstrap specific prop to a non reactstrap component.

```jsx
// user renders:
<DropdownToggler color="danger" tag="span" />
// DropdownToggler render method
<Button {...extraStuff} color="danger" tag="span"  />
// Button render method
<span className="..." />
```

I think on paper it's good but I haven't tried it just yet. If it seems reasonable could you update this component in that fashion?